lib/builder/builder.js

```diff
@@ -246,6 +249,10 @@ module.exports = {
 					tasks: projectTasks,
 					project,
 					parentLogger: log
+				}).then(async () => {
+					if (clearDest) {
```

Comment:
Also feedback from our internal discussion: It might make sense to move the dest removal part to *before* the build is started. I.e. Line [273](https://github.com/SAP/ui5-builder/pull/216/files#diff-f0272d7117ea8e9aba2b98d41e66e1a9R273) (feel free to make this function `async`).

Scenario: If the removal of the dist directory fails because of for example permission issues or because a file is still in use, a user would notice right away. With the current implementation, the user needs to wait for the whole build process to finish before receiving the error and the build is aborted.

Reply:
Good point.

Cleaning before the actual build runs means that the target folder is always cleared, even if the new build has failed. But I can live with that.